### PR TITLE
TOSA to TTIR refactor: split pass and patterns into separate files

### PR DESCRIPTION
**Pull request by sdjukicTT**
Split `lib/Conversion/TosaToTTIR/TosaToTTIR.cpp` into two files, `lib/Conversion/TosaToTTIR/TosaToTTIRPass.cpp` and `lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp`. This is done to prevent the file from becoming too large and difficult to read, as well as for consistency, to resemble how StableHLO to TTIR conversion is structured.

### CODE REVIEW

File: lib/Conversion/TosaToTTIR/TosaToTTIRPass.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #include "mlir/Transforms/DialectConversion.h"
 
 using namespace mlir;
-using namespace tt;
+using namespace mlir::tt;
```
Comment (sdjordjevicTT) [inline lib/Conversion/TosaToTTIR/TosaToTTIRPass.cpp:25]: You removed a lot of code from this file, are all of the above includes still required?

Comment (kmitrovicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPass.cpp:25]: If clang is properly set up, it grays out unused imports. Try removing them and see if compile and a couple of tests will work.

**Review by kmitrovicTT — APPROVED** [review]

File: lib/Conversion/TosaToTTIR/TosaToTTIRPass.cpp (hunk 1 above)
Comment (sdjukicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPass.cpp:21]: I have removed the unused includes.

Comment (sdjordjevicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPass.cpp:21]: Thanks!

**Review by sdjordjevicTT — APPROVED** [review]

File: lib/Conversion/TosaToTTIR/TosaToTTIRPass.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttmlir/Conversion/TosaToTTIR/TosaToTTIR.h"
+#include "ttmlir/Dialect/TTIR/IR/TTIR.h"
```
[the remark this answers is not in the document]
Comment (sdjukicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPass.cpp]: Moved it.

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttmlir/Conversion/TosaToTTIR/TosaToTTIR.h"
+#include "ttmlir/Dialect/TT/IR/TTOpsTypes.h"
+#include "ttmlir/Dialect/TTIR/IR/TTIROps.h"
```
[the remark this answers is not in the document]
Comment (sdjukicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp]: Moved it.

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+#include "mlir/Transforms/DialectConversion.h"
+
+using namespace mlir;
+using namespace mlir::tt;
+
+namespace {
+
+template <typename SrcOp, typename DestOp,
+          typename Adaptor = typename SrcOp::Adaptor>
+class TosaToTTIROpConversionPattern : public OpConversionPattern<SrcOp> {
```
[the remark this answers is not in the document]
Comment (sdjukicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp]: Renamed.

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  using OpConversionPattern<SrcOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(SrcOp srcOp, Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if constexpr (std::is_same<SrcOp, tosa::MulOp>::value) {
+      assert(srcOp.getShift() == 0);
+    }
```
[the remark this answers is not in the document]
Comment (sdjukicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp]: Added `checkConversionLegality`, a separate multiply op conversion pattern and a TODO comment.

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+    auto outputType = mlir::cast<RankedTensorType>(srcOp.getResult().getType());
```
[the remark this answers is not in the document]
Comment (sdjukicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp]: Fixed.

File: lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    auto outputTensor = rewriter.create<tensor::EmptyOp>(
```
[the remark this answers is not in the document]
Comment (sdjukicTT) [reply lib/Conversion/TosaToTTIR/TosaToTTIRPatterns.cpp]: Fixed.

**Pull request merged**